### PR TITLE
Military Overhaul: the Technologies

### PR DESCRIPTION
**Pull request by zumbak04**
Changelog:
- Skirmish Techniques and Melee Techniques are back.
- Balanced the first 4 technologies in such way that each of them gives buff to 3 basic units and their special units:
  - Frontline Troops give defensive and moral buff to Casters, Infantry and Ranged Infantry.
  - Assault Troops give defensive and moral buff to Large Infantry, Cavalry and Aerial Infantry.
  - Skirmish Techniques give offensive buff to Casters, Aerial Infantry and Ranged Infantry.
  - Melee Techniques give offensive buff to Infantry, Large Infantry and Cavalry.
- There's a few exceptions:
  - Siege Machines and War Machines get all buffs from Military Engineering.
  - Horse Archers get offensive buff from Skirmish Techniques instead of Melee Techniques.

### CODE REVIEW

File: localisation/000_wc_military_localization.csv [hunk 1]
```diff
@@ -1,17 +1,13 @@
 #CODE;ENGLISH;FRENCH;GERMAN;;SPANISH;;;;;;;;;x
 #Military Technologies;;;;;;;;;;;;;;x
-TECH_INFANTRY;Conventional Warfare;;Konventionelle Kriegsf�hrung;;;;;;;;;;;x
-TECH_CAVALRY;Specialized Warfare;;Irregul�re Kriegsf�hrung;;;;;;;;;;;x
-TECH_SKIRMISH;Ley Attunement;;Ley Einstimmung;;;;;;;;;;;x
-TECH_MELEE;Tactics Development;;Taktikentwicklung;;;;;;;;;;;x
+TECH_INFANTRY;Frontline Troops;;;;;;;;;;;;;x
+TECH_CAVALRY;Charging Troops;;;;;;;;;;;;;x
```
Comment (DukeJeans) [inline localisation/000_wc_military_localization.csv]: If we are using this route, perhaps `Shock Troops` instead?

File: localisation/000_wc_military_localization.csv [hunk 2]
```diff
@@ -1,17 +1,13 @@
...
 TECH_RECRUITMENT;Martial Logistics;;Kriegslogistik;;;;;;;;;;;x
-TECH_INFANTRY_DESC;Better equipment and discipline for our conventional troops;;Bessere Ausr�stung und Disziplin f�r unsere konventionellen Truppen;;;;;;;;;;;x
-TECH_CAVALRY_DESC;Better equipment and discipline for our more unconventional troops;;Bessere Ausr�stung und Disziplin f�r unsere Sondereinheiten;;;;;;;;;;;x
-TECH_SKIRMISH_DESC;Better training and discipline for our magically inclined troops;;Bessere Ausbildung und Disziplin f�r unsere magisch veranlagten Truppen;;;;;;;;;;;x
-TECH_MELEE_DESC;More developed and modern warfare tactics;;Entwickelte und moderne Kriegstaktiken;;;;;;;;;;;x
-TECH_SIEGE_EQUIPMENT_DESC;More advanced siege techniques and equipment for our machine troops;;Fortschrittlichere Belagerungstechniken und -Ausr�stung f�r unsere mechanischen Truppen;;;;;;;;;;;x
-TECH_RECRUITMENT_DESC;More developed logistic systems and recruitment methods;;Weiterentwickelte Logistiksysteme und Rekrutierungsmethoden;;;;;;;;;;;x
+TECH_INFANTRY_DESC;Better Equipment and Discipline for our Frontline Troops;;;;;;;;;;;;;x
+TECH_CAVALRY_DESC;Better Equipment and Discipline for our Charging Troops;;;;;;;;;;;;;x
```
Comment (DukeJeans) [inline localisation/000_wc_military_localization.csv]: If we are using this route, perhaps Shock Troops instead?

File: localisation/000_wc_military_localization.csv [hunk 3]
```diff
@@ -1,17 +1,13 @@
...
+TECH_SIEGE_EQUIPMENT_DESC;More Advanced Siege Techniques and Equipment for our Machine Troops;;Fortschrittlichere Belagerungstechniken und -Ausr�stung f�r unsere mechanischen Truppen;;;;;;;;;;;x
```
Comment (DukeJeans) [inline localisation/000_wc_military_localization.csv]:
```suggestion
TECH_SIEGE_EQUIPMENT_DESC;More Advanced Siege Techniques and Equipment for our Mechanical Troops;;Fortschrittlichere Belagerungstechniken und -Ausr�stung f�r unsere mechanischen Truppen;;;;;;;;;;;x
```

File: localisation/000_wc_military_localization.csv [hunk 4]
```diff
@@ -1,17 +1,13 @@
...
+TECH_RECRUITMENT_DESC;More Developed Logistic Systems and Recruitment Methods;;Weiterentwickelte Logistiksysteme und Rekrutierungsmethoden;;;;;;;;;;;x
```
Comment (DukeJeans) [inline localisation/000_wc_military_localization.csv]:
```suggestion
TECH_RECRUITMENT_DESC;More Developed Logistical Systems and Recruitment Methods;;Weiterentwickelte Logistiksysteme und Rekrutierungsmethoden;;;;;;;;;;;x
```

File: localisation/000_wc_military_localization.csv (hunk 1 above)
Comment (zumbak04) [reply localisation/000_wc_military_localization.csv]: Shocks Troops are already taken. The name of Aerial Infantry and Cavalry group.

File: localisation/000_wc_military_localization.csv [hunk 5]
```diff
@@ -1,17 +1,13 @@
 #CODE;ENGLISH;FRENCH;GERMAN;;SPANISH;;;;;;;;;x
 #Military Technologies;;;;;;;;;;;;;;x
-TECH_INFANTRY;Conventional Warfare;;Konventionelle Kriegsf�hrung;;;;;;;;;;;x
-TECH_CAVALRY;Specialized Warfare;;Irregul�re Kriegsf�hrung;;;;;;;;;;;x
-TECH_SKIRMISH;Ley Attunement;;Ley Einstimmung;;;;;;;;;;;x
-TECH_MELEE;Tactics Development;;Taktikentwicklung;;;;;;;;;;;x
+TECH_INFANTRY;Frontline Troops;;;;;;;;;;;;;x
```
Comment (DukeJeans) [inline localisation/000_wc_military_localization.csv]: I may not particularly say that Casters and Archers are "frontline". Maybe `Primary Troops`, `Baseline Troops`, or `Principal Troops`?

File: localisation/000_wc_military_localization.csv [hunk 6]
```diff
@@ -1,17 +1,13 @@
...
 TECH_SIEGE_EQUIPMENT;Military Engineering;;Wehrtechnik;;;;;;;;;;;x
 TECH_RECRUITMENT;Martial Logistics;;Kriegslogistik;;;;;;;;;;;x
-TECH_INFANTRY_DESC;Better equipment and discipline for our conventional troops;;Bessere Ausr�stung und Disziplin f�r unsere konventionellen Truppen;;;;;;;;;;;x
-TECH_CAVALRY_DESC;Better equipment and discipline for our more unconventional troops;;Bessere Ausr�stung und Disziplin f�r unsere Sondereinheiten;;;;;;;;;;;x
-TECH_SKIRMISH_DESC;Better training and discipline for our magically inclined troops;;Bessere Ausbildung und Disziplin f�r unsere magisch veranlagten Truppen;;;;;;;;;;;x
-TECH_MELEE_DESC;More developed and modern warfare tactics;;Entwickelte und moderne Kriegstaktiken;;;;;;;;;;;x
-TECH_SIEGE_EQUIPMENT_DESC;More advanced siege techniques and equipment for our machine troops;;Fortschrittlichere Belagerungstechniken und -Ausr�stung f�r unsere mechanischen Truppen;;;;;;;;;;;x
-TECH_RECRUITMENT_DESC;More developed logistic systems and recruitment methods;;Weiterentwickelte Logistiksysteme und Rekrutierungsmethoden;;;;;;;;;;;x
+TECH_INFANTRY_DESC;Better Equipment and Discipline for our Frontline Troops;;;;;;;;;;;;;x
```
Comment (DukeJeans) [inline localisation/000_wc_military_localization.csv]: I may not particularly say that Casters and Archers are "frontline". Maybe Primary Troops, Baseline Troops, or Principal Troops?

File: localisation/000_wc_military_localization.csv (hunk 1 above)
Comment (DukeJeans) [reply localisation/000_wc_military_localization.csv]: I don't understand the conflict, isn't this just naming a technology? Or does this show up in modifier tooltips?

Comment (zumbak04) [reply localisation/000_wc_military_localization.csv]: It sounds like this tech buffs only Shock Troops, while it also gives bonus to Large Infantry.

File: localisation/000_wc_military_localization.csv (hunk 5 above)
Comment (zumbak04) [reply localisation/000_wc_military_localization.csv]: I like Baseline... call Cavalry Primary Troops if you play nomad... meeeh.

File: localisation/000_wc_military_localization.csv (hunk 1 above)
Comment (DukeJeans) [reply localisation/000_wc_military_localization.csv]: How about then `Assault Troops`?

Comment (zumbak04) [reply localisation/000_wc_military_localization.csv]: Sounds fine!

**Pull request merged**